### PR TITLE
Azure footprint analysis - Diagrams

### PR DESCRIPTION
**Pull request by tinachoB**
|Related issue|
|---|
|Related issue: #9270 |

## Description

As part of wm_azure memory consumption analysis, a sequence diagram has been created as well as a Readme.md file explaining the feature itself.

image

## Tests

These are documentation changes only. No code changes has been made on this PR.

### CODE REVIEW

File: architecture/wm_azure/001-sequence.puml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+' Created by Wazuh, Inc. <info@wazuh.com>.
+' This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+@startuml wm_azure
+    main -> wmodules : wm_config()
+    wmodules -> config : ReadConfig()
+    config -> config : read_main_elements()
+    config -> config : Read_WModule()
+    config -> config : wm_azure_read()
+    config -> config : wm_azure_api_read()
```
Comment (jnasselle) [inline architecture/wm_azure/001-sequence.puml:11]: I think that using activation/desactivation feature (++,--) will allow to see the real call stack

File: architecture/wm_azure/001-sequence.puml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    loop
+        wm_azure -> wm_azure_storage : wm_azure_storage()
+        loop
+            wm_azure_storage -> wm_azure_storage : WM_AZURE_SCRIPT_PATH + commands
+            wm_azure_storage -> wm_azure_storage : wm_exec(WM_AZURE_SCRIPT_PATH + commands)
+        end
+    end
+    wm_azure -> wm_azure : SendMSG(ROOTCHECK_MQ)
+end
+@enduml
```
Comment (jnasselle) [inline architecture/wm_azure/001-sequence.puml:42]:
```suggestion
@enduml

```

**Review by jnasselle — CHANGES_REQUESTED** [review]

File: architecture/wm_azure/001-sequence.puml (hunk 1 above)
Comment (tinachoB) [reply architecture/wm_azure/001-sequence.puml:11]: Good catch!

File: architecture/wm_azure/001-sequence.puml (hunk 2 above)
Comment (tinachoB) [reply architecture/wm_azure/001-sequence.puml]: Will do

File: architecture/wm_azure/001-sequence.puml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@startuml wm_azure
+    main -> wmodules : wm_config()
+    wmodules -> config : ReadConfig()
+    config -> config : read_main_elements()
+    config -> config : Read_WModule()
+    config -> config : wm_azure_read()
+    config -> config : wm_azure_api_read()
+    wm_azure -> wm_azure_setup : wm_azure_setup(config)
+    wm_azure_setup -> wmodules : wm_state_io()
+loop
```
Comment (Dwordcito) [inline architecture/wm_azure/001-sequence.puml:18]: Please add the denote that the loop is part of the mainloop thread that comes from the execution of the wazuh module (aka wm_azure_main) anyway, something I see in the whole diagram is that it shows a tree of "how" calls was made, which is not correct.

the important thing to see in the sequence diagram is the "what" does, let's use descriptive text so as not to have to go to see the code to see what it does

**Review by Dwordcito — CHANGES_REQUESTED** [review]

**Review by jnasselle — APPROVED** [review]
LGTM!

**Review by konangc87 — APPROVED** [review]

File: architecture/wm_azure/001-sequence.puml (hunk 3 above)
Comment (tinachoB) [reply architecture/wm_azure/001-sequence.puml:18]: Got it. I'll change the approach of the diagram and upload it again.

**Review by Dwordcito — APPROVED** [review]

**Comment by Dwordcito** [comment]
@vikman90 review and merge, here we conclude.

**Pull request merged**